lib/codit/population/population.py

```diff
@@ -31,21 +32,26 @@ def form_groupings(self, group_size):
         return (random.sample(self.people, group_size) for _ in range(len(self.people)))
 
     def seed_infections(self, n_infected, disease, seed_periods=None):
-        seed_periods = seed_periods or disease.days_infectious
-        for p in random.sample(self.people, n_infected):
-            p.set_infected(disease)
-            stage = random.random() * seed_periods
-            while p.days_infected() < stage:
-                p.update_time()
-
-    def count_infectious(self):
-        return sum(p.infectious for p in self.people)
-
-    def count_infected(self):
-        return len(self.infected())
-
-    def infected(self):
-        return [p for p in self.people if (p.disease is not None or p.immune)]
+        if type(disease) is not set: disease = {disease}
+        for d in disease:
+            seed_periods = seed_periods or d.days_infectious
+            for p in random.sample(self.people, n_infected):
+                p.set_infected(d)
+                stage = random.random() * seed_periods
+                while p.days_infected() < stage:
+                    p.update_time()
+
+    def count_infectious(self, disease=None):
+        infected = self.infected(disease)
+        return sum(p.infectious for p in infected)
+
+    def count_infected(self, disease=None):
+        return len(self.infected(disease))
+
+    def infected(self, disease=None):
+        if disease is None:
+            return [p for p in self.people if p.immunities]
+        return [p for p in self.people if (disease in p.immunities)]
```

Comment:
I think it would be more convenient if we add 'and p.infected' in the if condition, as disease in p.immunities could be introduced by vaccines.